torchaudio/backend/_soundfile_backend.py

```diff
@@ -27,7 +63,8 @@ def info(filepath: str, format: Optional[str] = None) -> AudioMetaData:
         AudioMetaData: meta data of the given audio.
     """
     sinfo = soundfile.info(filepath)
-    return AudioMetaData(sinfo.samplerate, sinfo.frames, sinfo.channels)
+    bits_per_sample = SUBTYPE_TO_BITS_PER_SAMPLE[sinfo.subtype]
```

Comment:
If PySoundFile (and the underlying libsndfile) is updated and a new format, say is added, this will raise KeyError.
This makes me wonder if we should add something like `-1` for meaning unsupported (unexpected) format. like
`SUBTYPE_TO_BITS_PER_SAMPLE.get(sinfo.subtype, -1)`

There is a tradeoff between throwing error or silently ignoring such case.
Throwing error will prevent users who only needs to know other information from using the function, but the issue is more visible to users so that they can file an issue and we can add support.
Ignoring and substituting a placeholder makes it harder for users to notice such case, while the function will keep working if they do not need an accurate `bits_per_sample` for the new format.

What do you think?

(libsox would simply fail with `RunTimeError` if it does not recognize a format)

Reply:
Great point, thanks for noting!

I feel like an error might be a bit annoying for users as the bit depth is probably not such a crucial info (could be wrong on this). So I went for issuing a warning instead.

Regarding the placeholder value, I'm going with 0 instead of -1 as 0 is already used to mean "irrelevant or unknown" (this is documented in `AudioMetaData`).

I don't feel strongly about any of these points so I'm happy to amend according to your comments!

Reply:
> I feel like an error might be a bit annoying for users as the bit depth is probably not such a crucial info (could be wrong on this). So I went for issuing a warning instead.
> 
> Regarding the placeholder value, I'm going with 0 instead of -1 as 0 is already used to mean "irrelevant or unknown" (this is documented in `AudioMetaData`).
> 

Both points sound good to me. Thanks!